Sources/XCMetricsBackendLib/UploadMetrics/Repository/LogFileRepository.swift

```diff
@@ -48,10 +64,19 @@ struct LocalLogFileRepository: LogFileRepository {
         return tmp.fileURL
     }
 
-    func get(logURL: URL) throws -> URL {
+    func get(logURL: URL) throws -> LogFile {
         let tmp = try TemporaryFile(creatingTempDirectoryForFilename: "\(UUID().uuidString).xcactivitylog")
         try FileManager.default.copyItem(atPath: logURL.path, toPath: tmp.fileURL.path)
-        return tmp.fileURL
+        return LogFile(remoteURL: logURL, localURL: tmp.fileURL)
+    }
+
+    func delete(log: LogFile, wasProcessed: Bool) throws {
+        try FileManager.default.removeItem(at: log.localURL)
+
+        // Always remove local files
+        if wasProcessed {
+            try FileManager.default.removeItem(at: log.remoteURL)
```

Comment:
I thought `remoteURL` is an URL to some remote server, but I see we are deleting it from disc. 🤔 What's the difference between `localURL` and `remoteURL`?

Reply:
This  function is used in `LocalLogFileRepository` that is an implementation that uses the disk instead of a remote file storage.
In this case, remoteURL is the original local path where the log was stored when it was uploaded, and the localURL is a copy that is used for parsing the log. If there is an error, we retry 3 times before failing the job (that's done automatically by Vapor). So we can't delete the remoteURL because it won't be possible to process the log again, so we only delete the localURL. Only once the log has been processed, we delete the original log as well.